lib/internal/url.js

```diff
@@ -267,18 +259,22 @@ class URL {
 }
 
 Object.defineProperties(URL.prototype, {
-  toString: {
-    // https://heycam.github.io/webidl/#es-stringifier
-    writable: true,
-    enumerable: true,
-    configurable: true,
+  [kFormat]: {
+    enumerable: false,
+    configurable: false,
     // eslint-disable-next-line func-name-matching
-    value: function toString(options) {
+    value: function format(options) {
       options = options || {};
-      const fragment =
-        options.fragment !== undefined ?
-          !!options.fragment : true;
-      const unicode = !!options.unicode;
+      if (typeof options !== 'object')
+        throw new TypeError('options must be an object');
+      options.fragment = options.fragment !== undefined ?
+        Boolean(options.fragment) : true;
+      options.unicode = options.unicode !== undefined ?
+        Boolean(options.unicode) : false;
+      options.search = options.search !== undefined ?
+        Boolean(options.search) : true;
+      options.auth = options.auth !== undefined ?
+        Boolean(options.auth) : true;
```

Comment:
Not particularly a fan of Object.assign() due to some performance concerns that still exist there but this change should be fine